Extensions/NavMeshPathfindingBehavior/JsExtension.js

```diff
@@ -0,0 +1,782 @@
+    .setFunctionName('getDestinationX');
+
+  pathfindingBehaviorDeclaration
+    .addExpression(
+      'DestinationY',
+      _('Destination Y position'),
+      _('Destination Y position'),
+      _('Movement on the path'),
+      'CppPlatform/Extensions/AStaricon16.png'
+    )
+    .addParameter('object', _('Object'), '', false)
+    .addParameter(
+      'behavior',
+      _('Behavior'),
+      'NavMeshPathfindingBehavior',
+      false
+    )
+    .getCodeExtraInformation()
+    .setFunctionName('getDestinationY');
+
+  pathfindingBehaviorDeclaration
+    .addExpression(
+      'MovementAngle',
+      _('Angle of movement on its path'),
+      _('Angle of movement on its path'),
+      _('Movement on the path'),
+      'CppPlatform/Extensions/AStaricon16.png'
+    )
+    .addParameter('object', _('Object'), '', false)
+    .addParameter(
+      'behavior',
+      _('Behavior'),
+      'NavMeshPathfindingBehavior',
+      false
+    )
+    .getCodeExtraInformation()
+    .setFunctionName('getMovementAngle');
+};
+
+const declareObstacleBehavior = function (
+  _ /*: (string) => string */,
+  gd /*: libGDevelop */,
+  extension /*: gdPlatformExtension */
+) {
+  //TODO Add some kind of layer to select the obstacles for an object?
+  const pathfindingObstacleBehavior = new gd.BehaviorJsImplementation();
+  // $FlowExpectedError - ignore Flow warning as we're creating a behavior
+  pathfindingObstacleBehavior.updateProperty = function (
+    behaviorContent,
+    propertyName,
+    newValue
+  ) {
+    return false;
+  };
+  // $FlowExpectedError - ignore Flow warning as we're creating a behavior
+  pathfindingObstacleBehavior.getProperties = function (behaviorContent) {
+    const behaviorProperties = new gd.MapStringPropertyDescriptor();
+    return behaviorProperties;
+  };
+  // $FlowExpectedError - ignore Flow warning as we're creating a behavior
+  pathfindingObstacleBehavior.initializeContent = function (behaviorContent) {};
+  extension
+    .addBehavior(
+      'NavMeshPathfindingObstacleBehavior',
+      _('Obstacle for NavMesh pathfinding'),
+      'NavMeshPathfindingObstacleBehavior',
+      _('Flag the object as being an obstacle for pathfinding.'),
+      '',
+      'CppPlatform/Extensions/pathfindingobstacleicon.png',
+      'NavMeshPathfindingObstacleBehavior',
+      pathfindingObstacleBehavior,
+      new gd.BehaviorsSharedData()
+    )
+    .setIncludeFile(
+      'Extensions/NavMeshPathfindingBehavior/navmeshpathfindingobstacleruntimebehavior.js'
+    );
+};
+
+module.exports = {
+  createExtension: function (
+    _ /*: (string) => string */,
+    gd /*: libGDevelop */
+  ) {
+    const extension = new gd.PlatformExtension();
+    extension.setExtensionInformation(
+      'NavMeshPathfinding',
+      _('NavMesh Pathfinding behavior'),
+      _(
+        'Pathfinding allows to compute an efficient path for objects, avoiding obstacles on the way.'
+      ),
+      'D8H',
+      'MIT'
+    );
+
+    extension
```

Comment:
I think these properties would fit well in a "BehaviorSharedData"?

See here an example:
https://github.com/4ian/GDevelop/blob/f4512242e46084fc32dafe68e3f1a5ee5aa542e2/Extensions/ExampleJsExtension/JsExtension.js#L242-L272

Typically, this is used by the physics engine to store the gravity, the size of the virtual world. I think this would be well adapted here too - as it's not global to the project (but to a scene instead).

Reply:
I see, I've never used scene properties before. I'll do that.

Reply:
Can having both be useful?
For instance, the ViewPoint and CellSize will probably be the same for every levels. The game properties could be used to default some scene properties. The UI could display the value from the game properties as a hint in the scene properties fields using a new metadata.

Reply:
I think that's in this case a UX improvement to be done for the editor rather than something that the extension should support itself :)